### PR TITLE
[ComfyUI-OmniXPU] Add XPU median/nanmedian dim-reduction workaround

### PR DESCRIPTION
**Pull request by KristianZeng**
## Summary

On Intel XPU the dim-reduction kernel for `torch.median` / `torch.nanmedian` is pathologically slow — far behind both a full sort on the same device and the corresponding CUDA path — across reduction length, dim, memory layout,
value distribution and dtype (including integer types). The global (no-dim) median path is healthy. This adds a `ComfyUI-OmniXPU` patch (`median_fix`) that only intervenes on the XPU dim-reduction path and leaves everything else untouched.

This surfaced in the HiDream workflow (`nodes_hidream_o1.py`'s median blend), where the median step dominated runtime; with the patch it is no longer a hotspot.

## What changed

- `patches/patch_median.py` (new): wraps `torch.median` / `torch.nanmedian` on XPU
  - small N → Batcher odd-even min/max compare-exchange network
  - large N → `torch.sort`, take the lower-median element
  - **Values are bit-exact** vs `torch.median`.
  - Indices return a valid median index (a position whose value equals the median) via equality-argmax. `OMNIXPU_MEDIAN_STRICT_INDICES=1` forces `kthvalue` to reproduce `torch.median`'s exact tie-break indices.
- `config.py`, `patches/__init__.py`: wire it into the dispatcher as `median_fix`.
- `README.md`: document the patch and its env vars.

## Safety / fallbacks

- Only engages for **XPU** tensors on the **dim-reduction** path over float/int dtypes.
- NaN-containing slices, `out=` calls, and any exception fall back to the original implementation, preserving `torch.median`'s NaN-propagation semantics.
- Global median (`torch.median(x)` with no dim) is left completely untouched.
- On by default; disable with `OMNIXPU_MEDIAN_FIX=0`.

## How to test

```bash
# default: patch on
# verify startup log shows: [OmniXPU] median_fix: applied
# run a HiDream workflow and confirm correct output + no median hotspot

# regression / parity check:
python -c "import torch; x=torch.randn(33,128,128,device='xpu'); v,i=torch.median(x,dim=0)；print(v.shape, i.shape)"
```

### CODE REVIEW

File: omni/ComfyUI-OmniXPU/patches/patch_median.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
+import logging
+
+import torch
+
+log = logging.getLogger("ComfyUI-OmniXPU")
+
+# Patch torch.median / torch.nanmedian on XPU.
+#
+# Why: on Intel XPU the dim-reduction median/nanmedian kernel is pathologically
```
Comment (xiangyuT) [inline omni/ComfyUI-OmniXPU/patches/patch_median.py:9]: On Arc B60/B70 torch 2.10

**Review by xiangyuT — APPROVED** [review]
LGTM

File: omni/ComfyUI-OmniXPU/patches/patch_median.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        k = (n - 1) // 2 + 1
+        v, i = torch.kthvalue(moved, k, dim=0)
+    else:
+        if n <= SMALL_N_MAX:
+            v = _minmax_lower_median(moved)
+        else:
+            s, _ = torch.sort(moved, dim=0)
+            v = s[(n - 1) // 2]
+        # a valid median index: first position along dim holding the value
+        i = (moved == v.unsqueeze(0)).to(torch.uint8).argmax(dim=0)
```
[the remark this answers is not in the document]
Comment (KristianZeng) [reply omni/ComfyUI-OmniXPU/patches/patch_median.py]: Fixed in abb1226

File: omni/ComfyUI-OmniXPU/patches/patch_median.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            return torch.return_types.median((v, i))
+        except Exception as e:  # never break the graph
+            log.warning("[OmniXPU] median fast path failed (%s); fallback", e)
+            return _orig_median(input, dim, keepdim)
+
+    def _patched_nanmedian(input, dim=None, keepdim=False, *, out=None):
+        if out is not None or not _should_handle(input, dim):
+            if dim is None:
+                return _orig_nanmedian(input)
+            return _orig_nanmedian(input, dim, keepdim)
```
[the remark this answers is not in the document]
Comment (KristianZeng) [reply omni/ComfyUI-OmniXPU/patches/patch_median.py]: Fixed in abb1226
